controllers/csi/provisioner/reconciler.go

```diff
@@ -30,29 +31,43 @@ import (
 	"github.com/Dynatrace/dynatrace-operator/controllers/dynakube"
 	"github.com/Dynatrace/dynatrace-operator/dtclient"
 	"github.com/Dynatrace/dynatrace-operator/logger"
+	"github.com/go-logr/logr"
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	logDir     = "log"
+	dataDir    = "datastorage"
+	versionDir = "version"
+)
+
 var log = logger.NewDTLogger().WithName("provisioner")
 
 // OneAgentProvisioner reconciles a DynaKube object
 type OneAgentProvisioner struct {
-	client       client.Client
-	opts         dtcsi.CSIOptions
-	dtcBuildFunc dynakube.DynatraceClientFunc
+	client        client.Client
+	opts          dtcsi.CSIOptions
+	dtcBuildFunc  dynakube.DynatraceClientFunc
+	mkDirAllFunc  func(path string, perm fs.FileMode) error
+	writeFileFunc func(filename string, data []byte, perm fs.FileMode) error
+	readFileFunc  func(filename string) ([]byte, error)
```

Comment:
Instead of introducing these functions I strongly recommend to have a look at https://github.com/spf13/afero.